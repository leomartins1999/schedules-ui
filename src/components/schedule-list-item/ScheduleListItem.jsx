import ListGroup from "react-bootstrap/ListGroup";
import Badge from "react-bootstrap/Badge";

function getVariantForStatus(status) {
    switch (status) {
        case "IMPORTING": return "warning";
        case "CALCULATING": return "info";
        case "ERROR": return "danger";
        default: return null;
    };
}

function ScheduleListItem({ schedule }) {
    const { id, name, status } = schedule;

    const variant = getVariantForStatus(status);

    return (
        <ListGroup.Item
            className="d-flex justify-content-between align-items-center"
            href={`#schedules/${id}`}
            variant={variant}
            action
        >
            {name}
            <Badge variant="primary" pill>
                {status}
            </Badge>
        </ListGroup.Item >
    );
}

export default ScheduleListItem
