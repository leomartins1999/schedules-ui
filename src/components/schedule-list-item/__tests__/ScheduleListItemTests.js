import renderer from 'react-test-renderer';
import ScheduleListItem from "../ScheduleListItem"

it('renders a done schedule list item', () => {
    const schedule = {
        id: "1",
        name: "Done schedule",
        status: "DONE"
    }

    const tree = renderer
        .create(<ScheduleListItem schedule={schedule} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
})

it('renders an importing schedule list item', () => {
    const schedule = {
        id: "2",
        name: "Importing schedule",
        status: "IMPORTING"
    }

    const tree = renderer
        .create(<ScheduleListItem schedule={schedule} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
})

it('renders a calculating schedule list item', () => {
    const schedule = {
        id: "3",
        name: "Calculating schedule",
        status: "CALCULATING"
    }

    const tree = renderer
        .create(<ScheduleListItem schedule={schedule} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
})

it('renders an error schedule list item', () => {
    const schedule = {
        id: "4",
        name: "Error schedule",
        status: "ERROR"
    }

    const tree = renderer
        .create(<ScheduleListItem schedule={schedule} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
})
