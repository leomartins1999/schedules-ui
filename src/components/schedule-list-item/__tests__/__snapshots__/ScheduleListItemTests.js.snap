// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a calculating schedule list item 1`] = `
<a
  className="d-flex justify-content-between align-items-center list-group-item list-group-item-info list-group-item-action"
  href="#schedules/3"
  onClick={[Function]}
>
  Calculating schedule
  <span
    className="badge rounded-pill bg-primary"
    variant="primary"
  >
    CALCULATING
  </span>
</a>
`;

exports[`renders a done schedule list item 1`] = `
<a
  className="d-flex justify-content-between align-items-center list-group-item list-group-item-action"
  href="#schedules/1"
  onClick={[Function]}
>
  Done schedule
  <span
    className="badge rounded-pill bg-primary"
    variant="primary"
  >
    DONE
  </span>
</a>
`;

exports[`renders an error schedule list item 1`] = `
<a
  className="d-flex justify-content-between align-items-center list-group-item list-group-item-danger list-group-item-action"
  href="#schedules/4"
  onClick={[Function]}
>
  Error schedule
  <span
    className="badge rounded-pill bg-primary"
    variant="primary"
  >
    ERROR
  </span>
</a>
`;

exports[`renders an importing schedule list item 1`] = `
<a
  className="d-flex justify-content-between align-items-center list-group-item list-group-item-warning list-group-item-action"
  href="#schedules/2"
  onClick={[Function]}
>
  Importing schedule
  <span
    className="badge rounded-pill bg-primary"
    variant="primary"
  >
    IMPORTING
  </span>
</a>
`;
